### PR TITLE
Makes paper use the language system

### PR DESCRIPTION
**Pull request by Altoids1**
Robert paid me to do this.
image

## Overview

Similar to #7265, except it affects paper instead of PDAs.

This also affects papers pre-written and laid about the station; they will now have a specific language to them, most of the time.

### Coder Warning
I had to do some slightly-messy rewriting of how paper works in order to do this. ``info_links`` is now gone, and ``info`` now stores prewritten text which appears before any human-written text. ~~``info`` text has no language and is understood by all, for now.~~ ``info`` text has the language specified by ``infolang``. If no ``infolang`` is set, it defaults to being omnilingual and ergo legible by all.

### Changelog
:cl:  Altoids
rscadd: Things you write on paper are now in your default language, and cannot be read by those who don't know it!
/:cl:

### CODE REVIEW

**Comment by MegaEmpirical** [comment]
Love seeing paper get some love. Always needs more.

File: code/modules/paperwork/paper.dm [hunk 1]
```diff
@@ -22,8 +31,8 @@
 	max_integrity = 50
 	dog_fashion = /datum/dog_fashion/head
 
-	var/info		//What's actually written on the paper.
-	var/info_links	//A different version of the paper which includes html links at fields and EOF
+	var/info = "" // What's prewritten on the paper. Appears first and is a special snowflake callback to how paper used to work.
+	var/list/written = list()	//What's written on the paper by people. Stores /datum/langtext values, plus plaintext values that mark where fields are.
```
Comment (ThatLing) [inline code/modules/paperwork/paper.dm]: Assign this variable in /Initialize()

**Review by ThatLing — DISMISSED** [review]

File: code/modules/paperwork/paper.dm (hunk 1 above)
Comment (Altoids1) [reply code/modules/paperwork/paper.dm]: \>:(

**Comment by MegaEmpirical** [comment]
So will this work with books?

**Comment by Altoids1** [comment]
> So will this work with books?

It won't affect books, actually. I can do that in a separate PR, as I did with PDAs.

**Review by Hopekz — APPROVED** [review]

File: code/modules/paperwork/paper.dm [hunk 2]
```diff
@@ -5,6 +5,15 @@
  * lipstick wiping is in code/game/objects/items/weapons/cosmetics.dm!
  */
 
+#define PAPER_FIELD "<span class=\"paper_field\"></span>"
```
Comment (alexkar598) [inline code/modules/paperwork/paper.dm:8]: Since this define is only used in this file, undef it at the end of the file

Comment (Altoids1) [reply code/modules/paperwork/paper.dm:8]: BITCH I DID

Comment (ThatLing) [reply code/modules/paperwork/paper.dm:8]: undef it harder

**Pull request merged**